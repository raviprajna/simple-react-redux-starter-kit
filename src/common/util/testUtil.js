import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export function getComponentTree( component){
    const wrapper = shallow(component);
    const shallowComponent = wrapper.dive();
    return toJson(shallowComponent);
}
export function getWrapper( component){
    const wrapper = shallow(component);
    return wrapper;
}