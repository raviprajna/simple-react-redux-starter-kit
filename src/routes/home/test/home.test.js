import React from 'react';
import Home from '../';
import {getComponentTree} from '../../../common/util/testUtil.js'

describe('<Home/> Component Test', () => {

  it('renders <Home /> ', () => {
    expect(getComponentTree(<Home />)).toMatchSnapshot();
  });

});