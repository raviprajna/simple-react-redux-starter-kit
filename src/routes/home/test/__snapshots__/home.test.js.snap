// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Component Test renders <Home />  1`] = `
<StyledComponent
  flexDirection="column"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "display: flex;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
      },
      "displayName": "Styled(styled.div)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "propTypes": Object {
        "alignItems": [Function],
        "flexDirection": [Function],
        "flexWrap": [Function],
        "justifyContent": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <Styled(styled.div)
    as="a"
    color="blue"
    href="/reviews"
  >
    REVIEWS 
  </Styled(styled.div)>
  <Styled(styled.div)
    alt="First slide"
    as="img"
    m={0}
    p={3}
    src="Banner.png"
  />
</StyledComponent>
`;
